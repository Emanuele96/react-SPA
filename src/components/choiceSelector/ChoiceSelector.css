form > label{
    display: block; 
    padding: 5px;
    font-size: 1.2rem;
    
}

.choice-form{
    padding: 15px 0;
}
.choice-form > h3 {
    font-size: 1.5rem;
    
}

label > input {
    text-align: right; 
    background-color: red;
}

input {
    background-color: orange;
    margin-right: 5px;
}



