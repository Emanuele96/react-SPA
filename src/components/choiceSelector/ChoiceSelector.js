import React from 'react';

class ChoiceSelector extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedOption: this.props.alternative1
    };
    this.handleChange = this.handleChange.bind(this);
    this.props.sendCategory(this.state.selectedOption);
  }

  handleChange(e) {
    this.setState({
      selectedOption: e.target.value
    });
    this.props.sendCategory(e.target.value); //bug here, cannot pass this.state.selectedOption because is wrong. To fix
  }

  render() {
    return (
      <div className='ChoiceForm'>
        <h3>{this.props.categoryName}</h3>
        <form>
          <label>
            {this.props.alternative1}
            <input
              type='radio'
              checked={this.state.selectedOption === this.props.alternative1}
              value={this.props.alternative1}
              onChange={this.handleChange}
              name='category'
            />
          </label>

          <label>
            {this.props.alternative2}
            <input
              type='radio'
              checked={this.state.selectedOption === this.props.alternative2}
              value={this.props.alternative2}
              onChange={this.handleChange}
              name='category'
            />
          </label>

          <label>
            {this.props.alternative3}
            <input
              type='radio'
              checked={this.state.selectedOption === this.props.alternative3}
              value={this.props.alternative3}
              onChange={this.handleChange}
              name='category'
            />
          </label>
        </form>
      </div>
    );
  }
}

export default ChoiceSelector;
