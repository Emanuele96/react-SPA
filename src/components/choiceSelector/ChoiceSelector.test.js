import React from 'react';
import renderer from 'react-test-renderer';

import ChoiceSelector from './choiceSelector';



it('renders correctly ', () => {
    const tree = renderer.create(<ChoiceSelector sendCategory={()=> 0}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

