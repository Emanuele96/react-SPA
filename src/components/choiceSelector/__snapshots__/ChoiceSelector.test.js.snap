// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  1`] = `
<div
  className="choice-form"
>
  <h3 />
  <form>
    <label>
      <input
        checked={true}
        name="category"
        onChange={[Function]}
        type="radio"
      />
    </label>
    <label>
      <input
        checked={true}
        name="category"
        onChange={[Function]}
        type="radio"
      />
    </label>
    <label>
      <input
        checked={true}
        name="category"
        onChange={[Function]}
        type="radio"
      />
    </label>
  </form>
</div>
`;
